microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -0,0 +1,151 @@
+import org.eclipse.jdt.internal.core.manipulation.dom.ASTResolving;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.IJavaDiagnosticsParticipant;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+public class MicroProfileFaultToleranceDiagnosticsParticipant implements IJavaDiagnosticsParticipant {
+	
+	@Override
+	public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		IJavaProject javaProject = context.getJavaProject();
+		return JDTTypeUtils.findType(javaProject, FALLBACK_ANNOTATION) != null;
+	}
+
+	@Override
+	public List<Diagnostic> collectDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		ITypeRoot root = context.getTypeRoot();
+		IJavaElement[] elements = root.getChildren();
+		List<Diagnostic> diagnostics = new ArrayList<>();
+		collectDiagnostics(elements, diagnostics, context, monitor);
+		return diagnostics;
+	}
+
+	private void collectDiagnostics(IJavaElement[] elements, List<Diagnostic> diagnostics,
+			JavaDiagnosticsContext context, IProgressMonitor monitor) {
+		for (IJavaElement element : elements) {
+			if (monitor.isCanceled()) {
+				return;
+			} else if (element.getElementType() == IJavaElement.TYPE) {
+				IType type = (IType) element;
+				validateClass(type, diagnostics, context, monitor);
+			}
+		}
+	}
+
+	private void validateClass(IType type, List<Diagnostic> diagnostics, JavaDiagnosticsContext context,
+			IProgressMonitor monitor) {
+		Set<String> methodSet;
+		try {
+			methodSet = Stream.of(type.getMethods()).map(m -> {
+				return m.getElementName();
+			}).collect(Collectors.toUnmodifiableSet());
+		} catch (JavaModelException e1) {
+			return;
+		}
+		CompilationUnit ast = ASTResolving.createQuickFixAST(type.getCompilationUnit(), null);
+		ast.accept(new ASTVisitor() {
+
+			@Override
+			public boolean visit(MethodDeclaration node) {
+				validateMethod(node, methodSet, type, diagnostics, context, monitor);
+				super.visit(node);
+				return true;
+			}
+
+		});
+	}
+	
+	private void validateMethod(MethodDeclaration node, Set<String> methodSet, IType type, List<Diagnostic> diagnostics,
+			JavaDiagnosticsContext context, IProgressMonitor monitor) {
+		List modifiers = node.modifiers();
+		for (Object modifier : modifiers) {
+			if (modifier instanceof NormalAnnotation) {
+				NormalAnnotation annotation = (NormalAnnotation) modifier;
+				if (FALLBACK_ANNOTATION_ABBR.equals(annotation.getTypeName().getFullyQualifiedName())) {
+					Expression fallbackMethodExpr = getFallbackMethodValue(annotation);
+					if (fallbackMethodExpr != null) {
+						String fallbackMethodName = fallbackMethodExpr.toString();
+						fallbackMethodName = fallbackMethodName.substring(1, fallbackMethodName.length() - 1);
```

Comment:
Is it working if you type wrong Java code like:

 * `@Fallback(fallbackMethod = )`
 * `@Fallback(fallbackMethod = XXXXX)`